osidb/models.py

```diff
@@ -856,6 +856,27 @@ def _validate_special_handling_modules(self):
                     "are marked as special handling but flaw does not contain statement.",
                 )
 
+    def _validate_private_source_no_ack(self):
+        """
+        Checks that flaws with private sources have ACK.
+        """
+        if (source := FlawSource(self.source)) and source.is_private():
+            if self.meta.filter(type=FlawMeta.FlawMetaType.ACKNOWLEDGMENT).exists():
+                return
+
+            if source.is_public():
+                alert_text = f"Flaw source of type {source} can be public or private, "
+                alert_text += (
+                    "ensure that it is public since the Flaw has no acknowledgments."
+                )
+            else:
+                alert_text = f"Flaw has no acknoledgments but source of type {source} is private, "
+                alert_text += "include them in acknowledgments."
```

Comment:
nitpick: This is probably rather cosmetic by I wonder why are there two assignments instead just one eventually with a multi-line string. Maybe I am just more use to the other option :smile_cat: I am not sure about any advantages of one above the other.

Reply:
I don't have a strong preference on that, I update for a single assignment as suggested.